sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/IPartitionManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Responsible for dealing with the interaction with the chosen storage service.  A class implementing
+    ///   this interface must be able to create checkpoints and list/claim ownership.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   An instance of a class implementing this interface is provided by the user in the <see cref="EventProcessor" />
+    ///   constructor.
+    /// </remarks>
+    ///
+    public interface IPartitionManager
```

Comment:
So, this is an area of differing opinions.   We're encourage to (strongly) prefer abstract classes over interfaces to allow for better backwards compatibility;  you can add items to an abstract class without breaking current consumers, but any change to an interface is a breaking change.  The trade-off is, of course, that you can only inherit from a single base where you can have one class implement multiple interfaces.   

My questions are:
  - How do we see this used by developers?   Is there a strong case for them to have a partition manager as part of a class rather than being an wholly-contained and dedicated instance?

  - How sure are we that this interface is rock solid and we won't be looking to tweak it in any way post-GA?

_(on a personal level, I'm fairly neutral on the debate - but in this case, I think that I'd opt for an abstract class since its responsibility is both important and narrow)_

Reply:
- In general, I believe both scenarios would be desired, but I can't imagine a situation in which one of them should be enforced in order to have something working. It should be fairly simple to adapt.
- There's no way to be sure for now, specially because the public API is still under discussion and we may make changes based on users feedback.

We can keep it as an abstract class for this preview. If it's something developers are strongly against, we'll know.

Reply:
I'm good with keeping it as an interface for preview 2, but can we please open an issue to track revisiting and potentially making a change for preview 3?